### PR TITLE
NLU System: Model #2

### PR DESCRIPTION
**Pull request by csbanon**
This pull request contains the implementation of the final AI model. The following features are included:

* Model definition in PyTorch.
* Training loop code.
* Evaluation loop code.
* Prediction function (`predict()`) that returns top tag predictions.
* Parameters JSON file.

The model is performing well so far. To test the code, use the following steps:

**Training the AI**

1. Head to the AI directory: `ai`
2. Run the `train.py` file: `python train.py`
3. The AI will be trained and the accuracy will show.

**Getting Predictions**

1. Head to the AI directory: `ai`
2. Open the Python environment: `python`
3. Import the function: `from chatbot import predict` 
4. Call the function and print the results: `print(predict("<INPUT>"))`
5. The program should print a dictionary with the top-two tags per type.

### CODE REVIEW

**Review by grizzlyThresher — APPROVED** [review]
It all looks good to me! Can't wait to see it all integrated
